app/models/institution_builder.rb

```diff
@@ -0,0 +1,416 @@
+      FROM svas
+      WHERE institutions.cross = svas.cross AND svas.cross IS NOT NULL
+        AND institutions.version = #{version_number};
+    SQL
+
+    Institution.connection.update(str)
+  end
+
+  def self.add_vsoc(version_number)
+    columns = Vsoc::USE_COLUMNS.map(&:to_s).map { |col| %("#{col}" = vsocs.#{col}) }.join(', ')
+
+    str = <<-SQL
+      UPDATE institutions SET #{columns}
+      FROM vsocs
+      WHERE institutions.facility_code = vsocs.facility_code
+        AND institutions.version = #{version_number};
+    SQL
+
+    Institution.connection.update(str)
+  end
+
+  def self.add_eight_key(version_number)
+    str = <<-SQL
+      UPDATE institutions SET eight_keys = TRUE
+      FROM eight_keys
+      WHERE institutions.cross = eight_keys.cross
+        AND eight_keys.cross IS NOT NULL
+        AND institutions.version = #{version_number};
+    SQL
+
+    Institution.connection.update(str)
+  end
+
+  def self.add_accreditation(version_number)
+    # Set the accreditation_type according to the hierarchy hybrid < national < regional
+    # We include only those accreditation that are institutional and currently active.
+    str = <<-SQL
+      UPDATE institutions SET accreditation_type = CASE
+        WHEN at_types @> '{ regional }' THEN 'regional'
+        WHEN at_types @> '{ national }' THEN 'national'
+        WHEN at_types @> '{ hybrid }' THEN 'hybrid'
+        ELSE NULL
+      END
+      FROM (
+        SELECT "cross", array_agg(DISTINCT(accreditation_type)) AS at_types
+        FROM accreditations
+        WHERE "cross" IS NOT NULL
+          AND accreditation_type IS NOT NULL
+          AND periods LIKE '%current%'
+          AND csv_accreditation_type = 'institutional'
+        GROUP BY "cross") AS cross_type_arr
+      WHERE cross_type_arr.cross = institutions.cross
+        AND institutions.version = #{version_number};
+    SQL
+
+    Institution.connection.update(str)
+
+    # Set the accreditation_status according to the hierarchy probation < show cause aligned by
+    # accreditation_type
+    str = <<-SQL
+      UPDATE institutions SET accreditation_status = CASE
+        WHEN as_statuses @> '{ show cause }' THEN 'show cause'
+        WHEN as_statuses @> '{ probation }' THEN 'probation'
+        ELSE NULL
+      END
+      FROM (
+        SELECT "cross", accreditation_type, array_agg(DISTINCT(accreditation_status)) AS as_statuses
+        FROM accreditations
+        WHERE "cross" IS NOT NULL
+          AND accreditation_type IS NOT NULL
+          AND (accreditation_status = 'probation' OR accreditation_status = 'show cause')
+          AND periods LIKE '%current%'
+          AND csv_accreditation_type = 'institutional'
+        GROUP BY "cross", accreditation_type) AS cross_status_arr
+      WHERE institutions.cross = cross_status_arr.cross
+        AND institutions.accreditation_type = cross_status_arr.accreditation_type
+        AND institutions.version = #{version_number};
+    SQL
```

Comment:
Similar comment, as above. We need to look at these queries more closely. In particular, there is a ruby gem that helps with identifying database optimizations that can be made. I can't remember the name, but I will research and get back to you on that. That being said, these variety of optimizations could happen in a separate PR.